tests/test_annotated.py

```diff
@@ -470,3 +470,10 @@ class AnnotatedFieldModel4(BaseModel):
             'type': 'string_too_short',
         }
     ]
+
+
+def test_combine_annonated_fields():
+    class Foo(BaseModel):
+        v: Annotated[str, Field(json_schema_extra={'key1': 'value1'}), Field(json_schema_extra={'key2': 'value2'})]
+
+    assert {'key1': 'value1', 'key2': 'value2'}.items() <= Foo.model_json_schema()['properties']['v'].items()
```

Comment:
Good test case. However, original issue mentions usage of `TypeAdapter`, so perhaps add another case which specifically includes that.